File: 676-implement-magic-dictionary/implement-magic-dictionary.py
class MagicDictionary:

    def __init__(self):
        self.words = [] 
        

    def buildDict(self, dictionary: List[str]) -> None:
        for word in dictionary:
            self.words.append(word)

    def search(self, searchWord: str) -> bool:
        for word in self.words:
            if len(word) == len(searchWord):
                count = 0
                for i in range(len(word)):
                    if word[i] != searchWord[i]:
                        count+=1
                
                if count == 1: return True
        
        return False

        


# Your MagicDictionary object will be instantiated and called as such:
# obj = MagicDictionary()
# obj.buildDict(dictionary)
# param_2 = obj.search(searchWord)

File: 676-implement-magic-dictionary/implement-magic-dictionary.java
class MagicDictionary {
    List<String> dict;
    
    public MagicDictionary() {
        dict = new ArrayList();
    }
    
    public void buildDict(String[] dictionary) {
        for(String st: dictionary){
            dict.add(st);
        }
        
    }
    
    public boolean search(String searchWord) {
        for(int i = 0; i < dict.size();i++){
            if(dict.get(i).length()== searchWord.length()){
                String check = dict.get(i);
                int count = 0;
                for(int j = 0; j < searchWord.length();j++){
                    if(searchWord.charAt(j) != check.charAt(j)){ 
                        count+=1;
                    }
                }
                if(count == 1) return true;
            }
        }

        return false;
                
    }
}

/**
 * Your MagicDictionary object will be instantiated and called as such:
 * MagicDictionary obj = new MagicDictionary();
 * obj.buildDict(dictionary);
 * boolean param_2 = obj.search(searchWord);
 */